### DISCUSSION TITLE
Git Flow Proposal

### DISCUSSION THREAD

**Discussion by shoshta73** [gbowne1/json-maestro#38] [General]
I suggest branching off the current HEAD commit from `master` to `dev` branch.

We would use dev branch as main development branch, and when certain feature / improvement is ready to be merged we merge it to the master and if needed we tag the merge commit with associted versioned release

**Comment by shoshta73** [comment]
Also branch naming schema could use better documentation

**Comment by gbowne1** [comment]
I like Main and Master.

Main is a dev branch
Master is a permanent branch.  

From there you have bugfix and feature branches. 

What we really should do is proper release branches, so if someone wanted to use a particular version they could or if they wanted to bugfix a specific version they could 

Otherwise Git Flow would be appropriate

**Comment by shoshta73** [comment]
So if im following correctly, when we do release workflow, we should checkout the repo as is git switch at current ref, and push afterwards to the release/version branch?

If that is the case, that to big of an issue to implement but would require some changes in the release workflow. On the other git has built in feature to checkout/clone specific tag and or ref.

**Comment by gbowne1** [comment]
Idk.. lets wait and see.  I don't want to start just accumulating branches unless they are active. Merge branches can go away immediately.  Is there a workflow we can use that removes the branch once its complete merged and tested?

**Comment by gbowne1** [comment]
I don't think that having more than one dev branch and one production and or release branch is necessary.  Im ok with bugfix and feature branches.  

I like the idea of master being essential a stable backup
